timeseries/src/autogluon/timeseries/splitter.py

```diff
@@ -0,0 +1,192 @@
+                1970-01-04       4
+                1970-01-05       5
+                1970-01-06       6
+                1970-01-07       7
+                1970-01-08       8
+                1970-01-09       9
+                1970-01-10      10
+        >>> splitter = SlidingWindowSplitter(num_windows=2, overlap=1)
+        >>> train_data, val_data = splitter.split(ts_dataframe, prediction_length=3)
+        >>> print(train_data)
+                                target
+        item_id timestamp
+        0       1970-01-01       1
+                1970-01-02       2
+                1970-01-03       3
+                1970-01-04       4
+                1970-01-05       5
+        >>> print(val_data)
+                                    target
+        item_id       timestamp
+        0_[None:None] 1970-01-01       1
+                      1970-01-02       2
+                      1970-01-03       3
+                      1970-01-04       4
+                      1970-01-05       5
+                      1970-01-06       6
+                      1970-01-07       7
+                      1970-01-08       8
+                      1970-01-09       9
+                      1970-01-10      10
+        0_[None:-2]   1970-01-01       1
+                      1970-01-02       2
+                      1970-01-03       3
+                      1970-01-04       4
+                      1970-01-05       5
+                      1970-01-06       6
+                      1970-01-07       7
+                      1970-01-08       8
+    """
+
+    def __init__(self, num_windows: int, overlap: int = 0):
+        self.num_windows = num_windows
+        self.overlap = overlap
+
+    def describe_validation_strategy(self, prediction_length):
+        return (
+            f"Will use the last {self.num_windows} windows (each with prediction_length {prediction_length} "
+            f"time steps and overlap {self.overlap}) as a hold-out validation set."
+        )
+
+    def _split(
+        self, ts_dataframe: TimeSeriesDataFrame, prediction_length: int
+    ) -> Tuple[TimeSeriesDataFrame, TimeSeriesDataFrame]:
+        if self.overlap >= prediction_length:
+            raise ValueError(
+                f"SlidingWindowSplitter.overlap {self.overlap} must be < prediction_length {prediction_length}"
+            )
+        step_size = prediction_length - self.overlap
+
+        length_per_series = ts_dataframe.index.get_level_values(0).value_counts(sort=False)
+        num_total_validation_steps = prediction_length + step_size * (self.num_windows - 1)
+        num_too_short_series = (length_per_series <= num_total_validation_steps).sum()
+        if num_too_short_series > 0:
+            logger.warning(f"{num_too_short_series} are too short and won't appear in the training set")
+        if num_too_short_series == ts_dataframe.num_items:
+            raise ValueError(f"{self.name} produced an empty training set since all sequences are too short")
```

Comment:
Does it make sense to completely remove the series that are too short to produce `num_windows` many prediction intervals (as current version of the code does), or maybe we should make sure that, e.g., at least 50% of original series's length is preserved for training?

Reply:
I'm OK with this logic, but I can't see where the too short time series disappear. Does `slice_by_timestep` handle this gracefully?